### DISCUSSION TITLE
Question about intent to truncate simulation result

### DISCUSSION THREAD

**Discussion by sighpherdev** [biomass-dev/biomass#247] [Q&A] [answered]
Hi @himoto 
I would like to know the intention of saving the simulation results in the following code, where values less than `sys.float_info.epsilon` are considered as 0. 
In my model, the values of molecular concentration tend to be very small, with some molecules appearing with values around 1e-10. Among them, especially the smaller values are considered zero by `_preprocessing`. However, if I remove `_preprocessing` from the program and keep the values as they are, I get ideal results for simulation dynamics.
What is the problem with deleting this `_preprocessing`?

https://github.com/biomass-dev/biomass/blob/2ad16db84085b28f69ac41c439ec5e2660ac9120/biomass/dynamics/signaling_systems.py#L105-L116

**Comment by himoto** [comment, marked as answer]
When solving ODE numerically, sometimes we get infinitesimal value as a solution, e.g., 1e-308, although it should be zero.
This causes a problem when normalizing simulation results and `_preprocessing` can help avoid such problems.
It converts values to zeros only when all time-course values are smaller than `sys.float_info.epsilon`. Even if you delete this process and get something "ideal", the solution might be very sensitive to small changes in parameter values, and therefore I strongly recommend that you do not delete the process.

**Reply by sighpherdev** [reply]
So models that rely too much on small parameter changes should be avoided. I understand. Thank you very much.
